src/partitioned_data/partitionedNLPModels/partitionedNLPModel.jl

```diff
@@ -76,4 +76,44 @@ function NLPModels.grad!(
   return g
 end
 
+"""
+    B = hess_approx(nlp::PartiallySeparableNLPModel)
+
+Return the Hessian approximation of `nlp`.
+"""
+hess_approx(nlp::PartiallySeparableNLPModel) = get_pB(nlp)
+
+"""
+    Hx = hess(nlp, x; obj_weight=1.0)
+
+Evaluate the objective Hessian at `x` as a sparse matrix,
+with objective function scaled by `obj_weight`, i.e.,
+"""
+function NLPModels.hess(
+  nlp::PartiallySeparableNLPModel,
+  x::AbstractVector;
+  obj_weight=1.
+)
+  increment!(nlp, :neval_hess)
+  sp_hess = hess(nlp.part_data, x)
+  return obj_weight .* sp_hess
```

Comment:
You need to implement `hess_structure!` and `hess_coord!`, not `hess`.

Reply:
Remark: there is an interface to sparse matrices defined for any partitioned matrix, but it misses the symmetric part.

https://github.com/JuliaSmoothOptimizers/PartitionedStructures.jl/blob/69175afcbf6da33bfc5ac9f03a5198d932162b74/src/P_mat/part_mat.jl#L192